sycl/include/sycl/ext/oneapi/experimental/address_cast.hpp

```diff
@@ -63,39 +41,14 @@ multi_ptr<ElementType, Space, DecorateAddress> static_address_cast(
   if constexpr (Space == generic_space)
     return Ptr;
   else
-    return {static_address_cast<Space>(Ptr.get_raw())};
+    return {static_address_cast<Space>(Ptr.get_decorated())};
```

Comment:
We were limited by the extension interface before, so had to drop this bit of information in order to delegate. Now that we're calling something from `detail::` we can keep decoration in place.